ie_serving/models/shape_management/reshaper.py

```diff
@@ -0,0 +1,79 @@
+#
+# Copyright (c) 2019 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import json
+
+import falcon
+import datetime
+
+from ie_serving.logger import get_logger
+from ie_serving.models.shape_management.utils import ShapeMode
+from ie_serving.server.constants import GRPC, REST
+from ie_serving.server.predict_utils import statusCodes
+
+logger = get_logger(__name__)
+
+
+class Reshaper:
+
+    def __init__(self, service_type):
+        if service_type != GRPC and service_type != REST:
+            raise ValueError("Provided service type is unavailable")
+        self.service_type = service_type
+
+    def detect_shapes_incompatibility(self, engine, inference_input):
+        # Compares workload shapes with engine inputs shapes. If different,
+        # returns True, reshape_param.
+        # reshape_param is inputs shapes dictionary (input_name:shape pairs)
+        # for reshapable models and batch size for non-reshapable
+
+        reshape_required, inputs_shapes = engine.scan_input_shapes(
+            inference_input)
+        reshape_param = inputs_shapes
```

Comment:
Unnecessary assignment

Reply:
Here I assume that model is reshapable and by default reshape parameter will be dict of new input shapes. 
Right below, if model is not reshapable, I assign batch_size to reshape_param, so reshape_param can be either dict of shapes or integer depending on shape mode.